crypto/evp/pmeth_lib.c

```diff
@@ -137,6 +137,40 @@ EVP_PKEY_METHOD *EVP_PKEY_meth_new(int id, int flags)
 }
 #endif /* FIPS_MODULE */
 
+static int legacyalg(int id, const char *keytype)
+{
+#ifndef FIPS_MODULE
+    /* Certain EVP_PKEY keytypes are only available in legacy form */
+    if (id == -1) {
+        id = OBJ_sn2nid(keytype);
+        if (id == NID_undef)
+            id = OBJ_ln2nid(keytype);
+        if (id == NID_undef)
+            return  0;
+    }
+    switch (id) {
+    /*
+     * TODO(3.0): Remove SM2 and DHX when they are converted to have provider
+     * support
```

Comment:
I'm surprised to see the KDFs-by-PKEY here...  the *do* have provider support, all of them...  no?

Reply:
They have EVP_KDF support - but there is no EVP_KEYMGMT for them. The *legacy* implementation does the mapping to EVP_KDF. The same goes for the EVP_MAC support.